### PR TITLE
github/ci: Assorted refactoring, fixes and cleanups

### PR DESCRIPTION
**Pull request by phlax**
- Renames input/output vars to use hyphens rather than underscores
  I think this is more common, and generally preferable as it visually differentiates these kind of vars from others
- Consolidates related workflows
- Prepares prechecks for addition of further checks
- Replaces the original pre/post setup code in _ci wf with newer `steps-pre`/`steps-post`

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/actions/env/action.yml [hunk 1]
```diff
@@ -12,19 +12,19 @@ inputs:
...
     required: true
 
-  repo_ref:
+  repo-ref:
     type: string
-  repo_ref_sha:
+  repo-ref-sha:
     type: string
-  repo_ref_name:
+  repo-ref-name:
```
Comment (wbpcode) [inline .github/actions/env/action.yml:19]: could you add some context about why these changes are necessary? Seems there are lots of replacement from `_` to `-`. But why?

Comment (phlax) [reply .github/actions/env/action.yml:19]: added to PR description

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.
